test/pylib/scylla_cluster.py

```diff
@@ -277,9 +277,7 @@ async def cql_is_up(self) -> bool:
                          protocol_version=4,
                          auth_provider=auth) as cluster:
                 with cluster.connect() as session:
-                    session.execute("CREATE KEYSPACE IF NOT EXISTS k WITH REPLICATION = {" +
-                                    "'class' : 'SimpleStrategy', 'replication_factor' : 1 }")
-                    session.execute("DROP KEYSPACE k")
+                    session.execute("SELECT * FROM system.local")
```

Comment:
To just check if "CQL is up", you can do an even faster request, such as "BEGIN BATCH APPLY BATCH" (this is the command we use today in cql-pytest - see test/cql-pytest/conftest.py).

But I have a vague recollection that @kostja added this "CREATE KEYSPACE" thing precisely because there were points in time when CQL was responsive, but it was in fact started too early - at a point where CREATE KEYSPACE would fail (but SELECT for a system table would not). I'd like @kostja to please review this change and tell me if I'm mis-remembering, or maybe the situation regarding the boot order changed, or maybe this change is in fact wrong.

Reply:
By the way, I don't understand the commit message about this table creation being expensive because "we did this on every server restart". Server start is already a slow thing (taking roughly half a second if we're lucky), is table creation really a big chunk of that?

Reply:
The time it takes from starting the Scylla process to the moment you can perform a new schema change grows linearly with the number of schema changes previously performed - until a Raft group 0 snapshot is taken, and that takes 500 schema changes (IIRC - cc @gleb-cloudius) - because we must replay all existing committed Raft log entries before we can send and apply a new one.

Our tests don't perform enough schema changes to force a snapshot, I think (cc @kostja - perhaps we should have a test where snapshots are taken). In practice this means that the time to perform a "start server then make a schema change" operation grows linearly with the number of previous such operations. So it takes longer and longer to start servers and waiting for cql_is_up using this method.

Reply:
> But I have a vague recollection that @kostja added this "CREATE KEYSPACE" thing precisely because there were points in time when CQL was responsive, but it was in fact started too early - at a point where CREATE KEYSPACE would fail (but SELECT for a system table would not).

I have a vague recollection or something else: the problem was that the driver could connect (`cluster.connect()` successfully finished) but that wasn't enough, because CQL wasn't "really" available yet and attempting to perform a statement would fail. But I don't think it mattered what type of statement we try to perform.

Reply:
> ("USE SYSTEM" was the original one I used in cql-pytest and was what he saw at the time)

IIRC `USE` statements are purely driver-side - they don't communicate with the server (which kinda makes sense).
I can use `BEGIN BATCH APPLY BATCH`.

Reply:
> IIRC `USE` statements are purely driver-side - they don't communicate with the server (which kinda makes sense). 

This could have made sense, but I think it's not true in Cassandra on Scylla: We definitely have a "USE" statement in Cql.g, and cql3/statements/use_statement.cc that does something. The "USE SYSTEM" call I had in the past in cql-pytest worked well to distinguish working vs not-yet-up CQL. I changed it to `BEGIN BATCH APPLY BATCH` because it was annoying how it forced that silly USE setting on each connection, and I measured `BEGIN BATCH APPLY BATCH` to be equally do-nothing and quick.

In any case I'm still waiting for @kostja to tell us if he really needed the CREATE TABLE as an important operation (to somehow exercise schema operations) beyond the no-op stuff that you and I suggested.

Reply:
I talked to @kostja and apparently the reason for using CREATE/DROP was that Scylla needs to setup authorization, which it does after CQL is ready, which is required for schema changes to work.

I'm wondering though why would SELECT be allowed while CREATE/DROP keyspace wouldn't before this auth setup finishes.

Anyway, even if there was a problem with auth before I don't think we have it any longer because we now start Scylla with
```
    authenticator: AllowAllAuthenticator
    authorizer: AllowAllAuthorizer
```

I'm gonna switch to `BEGIN BATCH APPLY BATCH`. If auth comes up later we can perhaps come with a better solution.

Reply:
I checked in the code that allow all authorizer never looks at the system tables:
```                                                                                       
    virtual future<permission_set> authorize(const role_or_anonymous&, const resource&) const override {
        return make_ready_future<permission_set>(permissions::ALL);                                 
    }                                                                                               
```
          So it's safe to do anything if it's in use.

Reply:
Scylla permissions include:
```
typedef enum_set<                                                                                   
        super_enum<                                                                                 
                permission,                                                                         
                permission::READ,                                                                   
                permission::WRITE,                                                                  
                permission::CREATE,                                                                 
                permission::ALTER,                                                                  
                permission::DROP,                                                                   
                permission::SELECT,                                                                 
                permission::MODIFY,                                                                 
                permission::AUTHORIZE,                                                              
                permission::DESCRIBE>> permission_set; ```
                
   Batch statement class doesn't require or check permissions in itself, it relies on individual statements in the batch to check their permissions (modification statement). So an empty batch statement is a bad idea if a non-trivial authorizer is used.

Reply:
If the role is missing, NONE is returned as the permission set. So it should be enough to use pretty much any statement which requires a permission check as a safety net against races.